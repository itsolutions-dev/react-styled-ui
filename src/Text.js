import styled, { css } from './styledComponent';
import {
  getTextColor,
  getSizeOrDefault,
} from './utils/';

const Text = styled({
  component: 'p',
  style: css`
    font-size: ${props => getSizeOrDefault(props.size, props.theme.font.size)};
    font-family: ${props => props.theme.font.family};
    font-weight: ${props => props.bold === true ? 'bold' : props.theme.font.weight};
    line-height: ${props => getSizeOrDefault(props.lineHeight, props.theme.font.lineHeight)};
    ${(props) => {
      let align;
      if (props.left === true) align = 'left';
      if (props.right === true) align = 'right';
      if (props.center === true) align = 'center';
      if (props.justify === true) align = 'justify';
      if (align === undefined) return '';
      return `text-align: ${align};`;
    }}
    ${props => props.caps === true ? `
      text-transform: uppercase;
      letter-spacing: .2em;
    ` : ''}
    color: ${props => getTextColor(props)};
    margin: 0;
  `,
});

export default Text;
