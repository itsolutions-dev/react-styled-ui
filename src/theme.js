export default {
  radius: 8,
  spacing: {
    margin: 8,
    padding: 8,
    iconSize: 24,
    borderSize: 1,
    shadowSize: 1,
  },
  font: {
    size: 16,
    weight: 200,
    lineHeight: 18,
    family: 'Roboto, sans-serif',
  },
  h1: {
    size: 32,
    weight: 900,
    lineHeight: 64,
  },
  h2: {
    size: 24,
    weight: 700,
    lineHeight: 32,
  },
  h3: {
    size: 18,
    weight: 500,
    lineHeight: 24,
  },
  selected: {
    size: 0,
    weight: 10,
    lineHeight: 0,
  },
  hover: {
    size: 0,
    weight: 10,
    lineHeight: 0,
  },
  transitions: {
    in: undefined,
    out: undefined,
  },
  palette: {
    default: 'green',
    defaultText: 'white',
    defaultDisabled: undefined,
    defaultHover: undefined,
    defaultSelected: undefined,
    primary: 'red',
    primaryText: 'white',
    primaryDisabled: undefined,
    primaryHover: undefined,
    primarySelected: undefined,
    secondary: 'blue',
    secondaryText: 'white',
    secondaryDisabled: undefined,
    secondaryHover: undefined,
    secondarySelected: undefined,
    disabled: 'lightgray',
    shadow: 'black',
  },
};
