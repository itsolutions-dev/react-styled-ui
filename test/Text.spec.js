import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import 'styled-components-test-utils/lib/jest';
import theme from '../src/theme';
import Text from '../src/Text';

describe('Text', () => {
  test('should render a Text', () => {
    const component = ReactTestRenderer.create(<Text theme={theme} />);
    expect(component).toBeDefined();
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('should have a p tag', () => {
    const component = ReactTestRenderer.create(<Text theme={theme} />).toJSON();
    expect(component.type).toEqual('p');
  });
});
