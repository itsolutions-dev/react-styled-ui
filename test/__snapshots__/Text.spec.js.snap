// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should render a Text 1`] = `
.c0 {
  font-size: 12px;
  font-family: sans-serif;
  font-weight: 300;
  line-height: 18px;
  padding: 22px;
  margin: 0;
  color: #999;
}

<p
  className="c0"
/>
`;
